commit e9933f833bf336b107c2dc97bc1cc07d970cfb1c
Author: Christopher James Halse Rogers <christopher.halse.rogers@canonical.com>
Date:   Thu Jun 9 18:52:40 2011 +1000

    glx/dri2: Paper over errors in DRI2Connect when indirect
    
    DRI2 will throw BadRequest for this when the client is not local, but
    DRI2 is an implementation detail and not something callers should have
    to know about.  Silently swallow errors in this case, and just propagate
    the failure through DRI2Connect's return code.
    
    Bugzilla: https://bugs.freedesktop.org/show_bug.cgi?id=28125
    Signed-off-by: Christopher James Halse Rogers <christopher.halse.rogers@canonical.com>

diff --git a/src/glx/dri2.c b/src/glx/dri2.c
index adfd3d1..00adff2 100644
--- a/src/glx/dri2.c
+++ b/src/glx/dri2.c
@@ -180,6 +180,15 @@ DRI2Error(Display *display, xError *err, XExtCodes *codes, int *ret_code)
 	err->minorCode == X_DRI2DestroyDrawable)
 	return True;
 
+    /* If the server is non-local DRI2Connect will raise BadRequest.
+     * Swallow this so that DRI2Connect can signal this in its return code */
+    if (err->majorCode == codes->major_opcode &&
+        err->minorCode == X_DRI2Connect &&
+        err->errorCode == BadRequest) {
+	*ret_code = False;
+	return True;
+    }
+
     return False;
 }
 
